src/main/resources/reference.conf

```diff
@@ -243,15 +243,15 @@ divolte {
     // the Kafka producer.
     // See: http://kafka.apache.org/documentation.html#producerconfigs
     producer = {
-      metadata.broker.list = ["localhost:9092"]
-      metadata.broker.list = ${?DIVOLTE_KAFKA_BROKER_LIST}
-
+      bootstrap.servers = ["localhost:9092"]
+      bootstrap.servers = ${?DIVOLTE_KAFKA_BROKER_LIST}
       client.id = divolte.collector
       client.id = ${?DIVOLTE_KAFKA_CLIENT_ID}
 
-      request.required.acks = 0
-      message.send.max.retries = 5
-      retry.backoff.ms = 200
+      acks = 1
+      retries = 0
+      compression.type = lz4
+      max.in.flight.requests.per.connection = 1
```

Comment:
Compression is not impacted by the number of requests in flight. Note that `max.in.flight.requests.per.connection` controls the number of _requests_ that may be in-flight per connection with a broker, not the number of messages. The number of messages in the request will affect compression.

As things stand, these defaults yield:
 - Confidence messages have been delivered;
 - Less chance of reordering messages that relate to a given party in the face of temporary failures;
 - Protection against the Kafka producer blocking either due to metadata or the buffer being full.

As mentioned earlier, these are merely defaults. An administrator is free to prefer Kafka's retry mechanism, increase the concurrency, tweak `linger.ms`, adjust compression, and so on. There's nothing wrong with this, but the exact set of values are probably installation-specific. We're not telling you how to configure it for your purposes, but are providing a set of default values that should be safe and suitable for most users out-of-the-box.